dvc/system.py

```diff
@@ -21,55 +22,19 @@ def copy(src, dest):
 
     @staticmethod
     def hardlink(source, link_name):
-        import ctypes
-        from dvc.exceptions import DvcException
-
-        source, link_name = fspath(source), fspath(link_name)
-
-        if System.is_unix():
-            try:
-                os.link(source, link_name)
-                return
-            except Exception as exc:
-                raise DvcException("link") from exc
-
-        CreateHardLink = ctypes.windll.kernel32.CreateHardLinkW
-        CreateHardLink.argtypes = [
-            ctypes.c_wchar_p,
-            ctypes.c_wchar_p,
-            ctypes.c_void_p,
-        ]
-        CreateHardLink.restype = ctypes.wintypes.BOOL
-
-        res = CreateHardLink(link_name, source, None)
-        if res == 0:
-            raise DvcException("CreateHardLinkW") from ctypes.WinError()
+        try:
+            os.link(source, link_name)
+            return
+        except Exception as exc:
+            raise DvcException("failed to link") from exc
 
     @staticmethod
     def symlink(source, link_name):
-        import ctypes
-        from dvc.exceptions import DvcException
-
-        source, link_name = fspath(source), fspath(link_name)
-
-        if System.is_unix():
-            try:
-                os.symlink(source, link_name)
-                return
-            except Exception as exc:
-                msg = "failed to symlink '{}' -> '{}': {}"
-                raise DvcException(msg.format(source, link_name, str(exc)))
-
-        flags = 0
-        if source is not None and os.path.isdir(source):
-            flags = 1
-
-        func = ctypes.windll.kernel32.CreateSymbolicLinkW
-        func.argtypes = (ctypes.c_wchar_p, ctypes.c_wchar_p, ctypes.c_uint32)
-        func.restype = ctypes.c_ubyte
-
-        if func(link_name, source, flags) == 0:
-            raise DvcException("CreateSymbolicLinkW") from ctypes.WinError()
+        try:
+            os.symlink(source, link_name)
+            return
+        except Exception as exc:
+            raise DvcException("failed to symlink") from exc
```

Comment:
Previous exception had src and dst in it, which made it easier to debug. OSError that is raised doesn't have that info inside, right? Need to bring it back and probably do the same up above in hardlink. Btw, could you please replace `except Exception` with `except OSError`? Those os methods only raise those, so it shouldn've been replaced long ago, which is my bad 🙂

Reply:
Not sure how it works, but looks like the `'src' -> 'dst'` is already included in the errors.

```python
Python 3.8.1 (default, Dec 21 2019, 20:57:38)
[GCC 9.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.link('aoen', 'Modules')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'aoen' -> 'Modules'
>>> os.link('Modules', 'Doc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileExistsError: [Errno 17] File exists: 'Modules' -> 'Doc'
```

I prefer _not_ to be redundant: `ERROR: failed to symlink 'Modules' -> 'Doc' - File exists: 'Modules' -> 'Doc'`.


Although, I tried to read how `os.link` was implemented and couldn't find which exceptions are going to be raised from it, so, I'm not sure if this behavior is consistent.

As you mentioned before, `hardlink` just errors out with `DvcException("link")`, should I also add the `src` and `dst`?

Reply:
Looks like it is actually consistent:
```python
>>> import os
>>> os.link('/bin', 'here')
PermissionError: [Errno 1] Operation not permitted: '/bin' -> 'here'
```

So I'll remove them

Reply:
Oh, didn't know. Nice! Sure!